### PR TITLE
jenkins: start CI run on new pull requests

### PR DESCRIPTION
**Pull request by maclover7**
This will start a CI build on all new pull requests in nodejs/node, that
are opened by project collaborators

cc @joyeecheung

### CODE REVIEW

**Comment by refack** [comment]
FYI I just installed a plugin that can do the triggering (and also parse comments). 
https://wiki.jenkins.io/display/JENKINS/GitHub+pull+request+builder+plugin

Still need to bot to communicate back.

File: scripts/trigger-jenkins-build.js [hunk 1]
```diff
@@ -124,4 +148,33 @@ module.exports = (app) => {
...
+    const pullRequestAuthor = pull_request.user.login
+    const options = {
+      owner,
+      repo,
+      number,
+      logger
+    }
+
+    function replyToCollabWithBuildStarted (err, buildUrl) {
+      if (err) {
```
Comment (joyeecheung) [inline scripts/trigger-jenkins-build.js]: This does not actually reply to the thread right?

Comment (refack) [reply scripts/trigger-jenkins-build.js]: Why yes it does https://github.com/nodejs/node/pull/21045#issuecomment-393675826

Comment (maclover7) [reply scripts/trigger-jenkins-build.js]: @joyeecheung is right -- it originally commented, but then I removed the functionality (it relied on faulty Jenkins logic)

**Comment by refack** [comment]
I got our Jankins to do similar stuff
https://github.com/nodejs/node/pull/17330
I just need a token (or proxy it through the bot)

**Comment by devsnek** [comment]
is there a way we can disable this? sometimes people open WIP prs where running a ci would just be a waste

**Comment by maclover7** [comment]
@refack Interesting! Personally, I prefer having the code live inside of a repo somewhere, so we avoid the black box of Jenkins plugins

**Comment by maclover7** [comment]
@devsnek I don't believe so -- this purposely only runs `node-test-pull-request-lite` so it shouldn't take up many resources. Also, it only runs when PRs are first opened, so at worst it would only be one build that is automatically triggered

**Comment by refack** [comment]
The Jenkins plugin has a label "blacklist" so PR with the `WIP` label should be skipped

**Comment by Trott** [comment]
If we could get this to post a `CI: <URL here>` comment, that would rule.

**Comment by Trott** [comment]
Oh, I see that was attempted but ran into tech diff...

**Pull request merged**